### PR TITLE
Collect coverage metrics for all files

### PR DESCRIPTION
**Pull request by Gudahtt**
We now collect coverage metrics for all files, rather than just those executed by tests.

### CODE REVIEW

**Comment by Gudahtt** [comment]
Before:

```
----------------------|---------|----------|---------|---------|------------------------------------------------
File                  | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s                              
----------------------|---------|----------|---------|---------|------------------------------------------------
All files             |   87.01 |    79.65 |   94.87 |   86.86 |                                                
 changelog.js         |   77.11 |    63.27 |   91.67 |   76.83 | 16,188,190,192,198,220,224,226,237,253-270,287 
 constants.js         |     100 |      100 |     100 |     100 |                                                
 generateDiff.js      |     100 |      100 |     100 |     100 |                                                
 parseChangelog.js    |   89.74 |       75 |     100 |   89.74 | 30,45,60,75                                    
 validateChangelog.js |     100 |      100 |     100 |     100 |                                                
----------------------|---------|----------|---------|---------|------------------------------------------------

```

After:

```
----------------------|---------|----------|---------|---------|------------------------------------------------
File                  | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s                              
----------------------|---------|----------|---------|---------|------------------------------------------------
All files             |   41.29 |    44.55 |   56.92 |   41.08 |                                                
 changelog.js         |   77.11 |    63.27 |   91.67 |   76.83 | 16,188,190,192,198,220,224,226,237,253-270,287 
 cli.js               |       0 |        0 |       0 |       0 | 4-256                                          
 constants.js         |     100 |      100 |     100 |     100 |                                                
 generateDiff.js      |     100 |      100 |     100 |     100 |                                                
 index.js             |       0 |      100 |     100 |       0 | 1-7                                            
 parseChangelog.js    |   89.74 |       75 |     100 |   89.74 | 30,45,60,75                                    
 runCommand.js        |       0 |        0 |       0 |       0 | 1-79                                           
 updateChangelog.js   |       0 |        0 |       0 |       0 | 1-197                                          
 validateChangelog.js |     100 |      100 |     100 |     100 |                                                
----------------------|---------|----------|---------|---------|------------------------------------------------

```

File: jest.config.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 module.exports = {
   collectCoverage: true,
+  collectCoverageFrom: ['src/**/*.js', '!\\.test\\.js$'],
```
Comment (rekmarks) [inline jest.config.js:3]: Out of curiosity, doesn't this work?
```suggestion
  collectCoverageFrom: ['src/**/*.js', '!**/*test.js'],
```

**Review by rekmarks — DISMISSED** [review]

File: jest.config.js (hunk 1 above)
Comment (Gudahtt) [reply jest.config.js:3]: Good question! I will check

Comment (Gudahtt) [reply jest.config.js]: Apparently that entry wasn't needed at all! Jest automatically skips tests from coverage reporting. I've removed this in the latest commit.

File: jest.config.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 module.exports = {
   collectCoverage: true,
+  collectCoverageFrom: ['src/**/*\\.js'],
```
Comment (rekmarks) [inline jest.config.js:3]: So we do need to escape the period? 👀

**Review by rekmarks — APPROVED** [review]

File: jest.config.js (hunk 2 above)
Comment (Gudahtt) [reply jest.config.js:3]: Ah, whoops. I had added the escape here to be consistent with the `testRegex` below, but the Jest docs indicate that `collectCoverageFrom` is using [micromatch](https://github.com/micromatch/micromatch) but `testRegex`is not (presumably it's a regular RegExp :shrug: ). So the escaping is needed for `testRegex`, but not here. I'll remove it!

**Review by rekmarks — APPROVED** [review]

**Pull request merged**